bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/CompactionTest.java

```diff
@@ -332,6 +332,95 @@ public void checkpointComplete(Checkpoint checkPoint, boolean compact)
         });
     }
 
+    @Test
+    public void testForceGarbageCollectionWhenDiskIsFull() throws Exception {
+        testForceGarbageCollectionWhenDiskIsFull(true);
+        testForceGarbageCollectionWhenDiskIsFull(false);
+    }
+
+    public void testForceGarbageCollectionWhenDiskIsFull(boolean isForceCompactionAllowWhenDisableCompaction)
+        throws Exception {
+
+        restartBookies(conf -> {
+            if (isForceCompactionAllowWhenDisableCompaction) {
+                conf.setMinorCompactionInterval(0);
+                conf.setMajorCompactionInterval(0);
+                conf.setForceAllowCompaction(true);
+                conf.setMajorCompactionThreshold(0.5f);
+                conf.setMinorCompactionThreshold(0.2f);
+            } else {
+                conf.setMinorCompactionInterval(120000);
```

Comment:
can we use lower interval? this test will take several minutes to complete

Reply:
I just call the `triggerGC` to trigger compaction, so the interval won't affect the time cost of the test run.